File: app/src/main/java/com/example/sertac/renkbelirleyici/MainActivity.java
package com.example.sertac.renkbelirleyici;

import android.content.pm.ActivityInfo;
import android.graphics.Color;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.SeekBar;

public class MainActivity extends AppCompatActivity implements SeekBar.OnSeekBarChangeListener {

    private LinearLayout colorShower;
    private SeekBar redColor,greenColor,blueColor;
    private EditText redColorEdittext,greenColorEdittext,blueColorEdittext;
    int red,green,blue;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        init();
    }

    private void init() {
        colorShower= (LinearLayout) findViewById(R.id.colorShower);
        redColor= (SeekBar) findViewById(R.id.redColorSeekbar);
            redColor.setMax(255);
        greenColor= (SeekBar) findViewById(R.id.greenColorSeekbar);
            greenColor.setMax(255);
        blueColor= (SeekBar) findViewById(R.id.blueColorSeekbar);
            blueColor.setMax(255);
        redColorEdittext= (EditText) findViewById(R.id.redColorEdittext);
            redColorEdittext.setText(String.valueOf(redColor.getProgress()));
        greenColorEdittext= (EditText) findViewById(R.id.greenColorEdittext);
            greenColorEdittext.setText(String.valueOf(greenColor.getProgress()));
        blueColorEdittext= (EditText) findViewById(R.id.blueColorEdittext);
            blueColorEdittext.setText(String.valueOf(blueColor.getProgress()));
        actionEvent();
    }

    private void actionEvent() {
    redColor.setOnSeekBarChangeListener(this);
    blueColor.setOnSeekBarChangeListener(this);
    greenColor.setOnSeekBarChangeListener(this);
    redColorEdittext.addTextChangedListener(new TextWatcher() {
        @Override
        public void beforeTextChanged(CharSequence s, int start, int count, int after) {
        }

        @Override
        public void onTextChanged(CharSequence s, int start, int before, int count) {
            red=Integer.parseInt(String.valueOf(s));
            redColor.setProgress(red);
            backgroundChanger(red,green,blue);
        }

        @Override
        public void afterTextChanged(Editable s) {
        }
    });
    greenColorEdittext.addTextChangedListener(new TextWatcher() {
        @Override
        public void beforeTextChanged(CharSequence s, int start, int count, int after) {

        }

        @Override
        public void onTextChanged(CharSequence s, int start, int before, int count) {
                green=Integer.parseInt(String.valueOf(s));
                greenColor.setProgress(green);
                backgroundChanger(red,green,blue);
        }

        @Override
        public void afterTextChanged(Editable s) {

        }
    });
    blueColorEdittext.addTextChangedListener(new TextWatcher() {
        @Override
        public void beforeTextChanged(CharSequence s, int start, int count, int after) {

        }

        @Override
        public void onTextChanged(CharSequence s, int start, int before, int count) {
                blue=Integer.parseInt(String.valueOf(s));
                blueColor.setProgress(blue);
                backgroundChanger(red,green,blue);
        }

        @Override
        public void afterTextChanged(Editable s) {

        }
    });
        backgroundChanger(red,green,blue);
    }

    @Override
    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
        switch (seekBar.getId()){
            case R.id.redColorSeekbar:
                redColorEdittext.setText(String.valueOf(progress));
                break;
            case R.id.greenColorSeekbar:
                greenColorEdittext.setText(String.valueOf(progress));
                break;
            case R.id.blueColorSeekbar:
                blueColorEdittext.setText(String.valueOf(progress));
                break;
        }
    }

    @Override
    public void onStartTrackingTouch(SeekBar seekBar) {

    }

    @Override
    public void onStopTrackingTouch(SeekBar seekBar) {

    }

    private void backgroundChanger(int red,int green,int blue){
        colorShower.setBackgroundColor(Color.rgb(red,green,blue));
    }
}

File: app/src/main/java/com/example/sertac/renkbelirleyici/MainActivity.kt
package com.example.sertac.renkbelirleyici

import android.graphics.Color
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.Editable
import android.text.TextUtils
import android.text.TextWatcher
import android.view.View
import android.widget.SeekBar
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), SeekBar.OnSeekBarChangeListener,View.OnClickListener {

    private var red: Int = 0
    private var green: Int = 0
    private var blue: Int = 0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        redColorSeekbar.max = 255
        greenColorSeekbar.max = 255
        blueColorSeekbar.max = 255

        init()
    }

    private fun init() {
        redColorValue.setText(redColorSeekbar.progress.toString())
        greenColorValue.setText(greenColorSeekbar.progress.toString())
        blueColorValue.setText(blueColorSeekbar.progress.toString())
        actionEvent()
    }

    private fun actionEvent() {
        redColorSeekbar.setOnSeekBarChangeListener(this)
        blueColorSeekbar.setOnSeekBarChangeListener(this)
        greenColorSeekbar.setOnSeekBarChangeListener(this)
        redColorValue.setOnClickListener(this)
        greenColorValue.setOnClickListener(this)
        blueColorValue.setOnClickListener(this)

        redColorValue.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

            }

            override fun afterTextChanged(s: Editable) {
                red = Integer.parseInt(if (TextUtils.isEmpty(s)) "0" else s.toString())
                redColorSeekbar.progress = red
                backgroundChanger(red, green, blue)
            }
        })
        greenColorValue.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

            }

            override fun afterTextChanged(s: Editable) {
                green = Integer.parseInt(if (TextUtils.isEmpty(s)) "0" else s.toString())
                greenColorSeekbar.progress = green
                backgroundChanger(red, green, blue)
            }
        })
        blueColorValue.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

            }

            override fun afterTextChanged(s: Editable) {
                blue = Integer.parseInt(if (TextUtils.isEmpty(s)) "0" else s.toString())
                blueColorSeekbar.progress = blue
                backgroundChanger(red, green, blue)
            }
        })
        backgroundChanger(red, green, blue)
    }

    override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
        when (seekBar.id) {
            R.id.redColorSeekbar -> {
                redColorValue.setText(progress.toString())
            }
            R.id.greenColorSeekbar -> {
                greenColorValue.setText(progress.toString())
            }
            R.id.blueColorSeekbar -> {
                blueColorValue.setText(progress.toString())
            }
        }
    }

    override fun onStartTrackingTouch(seekBar: SeekBar) {
        when (seekBar.id){
            R.id.redColorSeekbar -> if (redColorValue.isFocusable) redColorValue.isFocusableInTouchMode=false

            R.id.greenColorSeekbar ->if (greenColorValue.isFocusable) greenColorValue.isFocusableInTouchMode=false

            R.id.blueColorSeekbar ->if (blueColorValue.isFocusable) blueColorValue.isFocusableInTouchMode=false

        }
    }

    override fun onStopTrackingTouch(seekBar: SeekBar) {

    }

    private fun backgroundChanger(red: Int, green: Int, blue: Int) {
        colorShower.setBackgroundColor(Color.rgb(red, green, blue))
        val value="# ${hexValue(red)}${hexValue(green)}${hexValue(blue)}"
        hex_color_value.text=value
    }

    override fun onClick(v: View) {
        if (!v.isFocusable) v.isFocusableInTouchMode=true
    }

    private fun hexValue(value:Int):String{
        return ""
    }
}
